src/screens/StudyRegisterScreen/StudyRegisterScreen.tsx

```diff
@@ -0,0 +1,94 @@
+import { FC, useState } from 'react';
+import { Box, Progress } from '@chakra-ui/react';
+import { useRecoilState } from 'recoil';
+import ScreenHeader from 'components/ScreenHeader';
+import SvgIcon from '@components/SvgIcon';
+import { RegisterFormData } from '../../../types/register';
+import { registerFormDataAtom } from './atoms';
+import { REGISTER_STEPS } from './consts';
+import {
+  InfoSettingStep,
+  NotiSettingStep,
+  RuleSettingStep,
+  ScheduleSettingStep,
+} from './steps';
+
+const StudyRegisterScreen: FC = () => {
+  const [step, setStep] = useState<(typeof REGISTER_STEPS)[number]>(
+    REGISTER_STEPS[0],
+  );
+
+  const [registerFormData, setRegisterFormData] =
+    useRecoilState(registerFormDataAtom);
+
+  console.log(
+    '🚀 ~ file: StudyRegisterScreen.tsx:22 ~ registerFormData:',
+    registerFormData,
+  );
+
+  const currentStepIdx = REGISTER_STEPS.findIndex((t) => t === step);
+
+  const handleClickPrev = () => {
+    if (currentStepIdx === 0) {
+      // TODO: 웹뷰 종료
+      return;
+    }
+    setStep(REGISTER_STEPS[currentStepIdx - 1]);
+  };
+
+  function handleChangeFormData(data: Partial<RegisterFormData>): void {
+    setRegisterFormData((prev) => ({ ...prev, ...data }));
+  }
+
+  return (
+    <Box>
+      <ScreenHeader
+        title="스터디 생성"
+        leftIcon={
+          <SvgIcon name="ArrowLeft" size={24} onClick={handleClickPrev} />
+        }
+      />
+      <Progress
+        value={(100 / REGISTER_STEPS.length) * (currentStepIdx + 1)}
+        height="3px"
+      />
+
+      <Box>
+        {step === 'info' && (
+          <InfoSettingStep
+            onNext={(data) => {
+              setStep('schedule');
+              handleChangeFormData(data);
+            }}
+          />
+        )}
+        {step === 'schedule' && (
+          <ScheduleSettingStep
+            onNext={(data) => {
+              setStep('rule');
+              handleChangeFormData(data);
+            }}
+          />
+        )}
+        {step === 'rule' && (
+          <RuleSettingStep
+            onNext={(data) => {
+              setStep('noti');
+              handleChangeFormData(data);
+            }}
+          />
+        )}
+        {step === 'noti' && (
+          <NotiSettingStep
+            onNext={(data) => {
+              setStep('info');
+              handleChangeFormData(data);
+            }}
+          />
+        )}
+      </Box>
```

Comment:
P2
이 부분은 이렇게도 표현할 수 있을 것 같은데 어떻게 생각해? (위 코멘트에서 Contents 부분과 함께)
간결하고 직관적이고 React스러운 방식이라고 생각해서 내가 다중 조건부에서 자주 쓰는 방식이야!


```suggestion
      <Box>
        <Contents registerStep={step} />
      </Box>
```

Reply:
답이 늦엇군ㅠ
이렇게 맵 이용한 설계는 나도 아주 좋아해~
다만 Content로 컴포넌트를 분리하는건 내가 의도햇던거와 달라지는듯 한데 나는 Screen컴포넌트에서 각 스텝마다의 onNext 모두 관리할 수 있으면 좋겠다고 생각했어!

그렇다고 컴포넌트 내에서 컴포넌트를 선언하는건 안티패턴이니.. 컴포넌트 분리는 하지 않은 상태에서 제안해준 맵방식으로 리팩토링 하는쪽으로 수정하면 어떨까?